subworkflows/local/merge_libraries.nf

```diff
@@ -0,0 +1,53 @@
+//
+// Merge libraries of the same sample, then sort, index, and flagstat the merged bam
+//
+
+include { SAMTOOLS_MERGE    as SAMTOOLS_MERGE_LIBRARIES           } from '../../modules/nf-core/samtools/merge/main'
+include { SAMTOOLS_SORT     as SAMTOOLS_SORT_MERGED_LIBRARIES     } from '../../modules/nf-core/samtools/sort/main'
+include { SAMTOOLS_INDEX    as SAMTOOLS_INDEX_MERGED_LIBRARIES    } from '../../modules/nf-core/samtools/index/main'
+include { SAMTOOLS_FLAGSTAT as SAMTOOLS_FLAGSTAT_MERGED_LIBRARIES } from '../../modules/nf-core/samtools/flagstat/main'
+
+workflow MERGE_LIBRARIES {
+    take:
+    ch_bam_bai  // [ [ meta ], bam , bai ]
+
+    main:
+    ch_versions      = Channel.empty()
+    ch_multiqc_files = Channel.empty()
+
+    ch_library_merge_input = ch_bam_bai
+        // TODO add 'id_index' to final meta? (once that also gets added to bam input). Maybe also keep SE/PE? (for now, we assume SE, see comment below)
```

Comment:
Still necessary?

Reply:
BAM input would ideally get an `id_index`, just like FastQs do. I think the SE/PE bit doesn't really matter at this point, so hard-coding should be ok. I made an issue about the metas of BAM inputs #1040 
Deleteing the comment, though we should revisit this once bams have consistent meta info.